### PR TITLE
Update MLTransform docs

### PR DESCRIPTION
**Pull request by AnandInguva**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by AnandInguva** [comment]
R: @rszper

File: website/www/site/content/en/documentation/ml/preprocess-data.md [hunk 1]
```diff
@@ -53,7 +50,10 @@ modules for machine learning tasks.
 ## Transforms {#transforms}
 
 You can use `MLTransform` to perform the following data processing transforms.
-For information about the transforms, see
+
+### Data Processing Transforms using TFT
```
Comment (rszper) [inline website/www/site/content/en/documentation/ml/preprocess-data.md:54]:
```suggestion
### Data processing transforms that use TFT
```

File: website/www/site/content/en/documentation/ml/preprocess-data.md [hunk 2]
```diff
@@ -53,7 +50,10 @@ modules for machine learning tasks.
 ## Transforms {#transforms}
 
 You can use `MLTransform` to perform the following data processing transforms.
```
Comment (rszper) [inline website/www/site/content/en/documentation/ml/preprocess-data.md:52]:
```suggestion
You can use `MLTransform` to perform various data processing transforms.
```

File: website/www/site/content/en/documentation/ml/preprocess-data.md [hunk 3]
```diff
@@ -53,7 +50,10 @@ modules for machine learning tasks.
...
-For information about the transforms, see
+
+### Data Processing Transforms using TFT
+
+For information about the tft based transforms, see
```
Comment (rszper) [inline website/www/site/content/en/documentation/ml/preprocess-data.md:56]:
```suggestion
The following set of transforms available in the `MLTransform` class come from
the TensorFlow Transforms (TFT) library. TFT offers specialized processing
modules for machine learning tasks. For information about these transforms, see
```

File: website/www/site/content/en/documentation/ml/preprocess-data.md [hunk 4]
```diff
@@ -70,9 +70,13 @@ TensorFlow documentation.
 | TFIDF | See [`tft.tfidf`](https://www.tensorflow.org/tfx/transform/api_docs/python/tft/tfidf) in the TensorFlow documentation. |:
 {{< /table >}}
 
-Apply the transforms on either single or multiple columns passed as a
-`dict` on structured data. Keys are column names and values are lists containing
-each column's data.
+### Generate Text Embeddings
```
Comment (rszper) [inline website/www/site/content/en/documentation/ml/preprocess-data.md:73]:
```suggestion
### Text embedding transforms
```

File: website/www/site/content/en/documentation/ml/preprocess-data.md [hunk 5]
```diff
@@ -70,9 +70,13 @@ TensorFlow documentation.
...
+
+{{< table >}}
+| Transform name | Description |
+| ------- | ---------------|
+| SentenceTransformerEmbeddings | Uses [sentence-transformer](https://huggingface.co/sentence-transformers) models to generate text embeddings. sentence-transformers models hosted on HuggingFace hub are supported.
```
Comment (rszper) [inline website/www/site/content/en/documentation/ml/preprocess-data.md:78]:
```suggestion
| SentenceTransformerEmbeddings | Uses the Hugging Face [`sentence-transformers`](https://huggingface.co/sentence-transformers) models to generate text embeddings.
```

File: website/www/site/content/en/documentation/ml/preprocess-data.md [hunk 6]
```diff
@@ -70,9 +70,13 @@ TensorFlow documentation.
...
+| VertexAITextEmbeddings | Uses [Vertex AI]( https://cloud.google.com/vertex-ai/docs/generative-ai/embeddings/get-text-embeddings) text embedding models to generate text embeddings.
```
Comment (rszper) [inline website/www/site/content/en/documentation/ml/preprocess-data.md:79]:
```suggestion
| VertexAITextEmbeddings | Uses models from the [the Vertex AI text-embeddings API](https://cloud.google.com/vertex-ai/docs/generative-ai/embeddings/get-text-embeddings) to generate text embeddings.
```

File: website/www/site/content/en/documentation/ml/preprocess-data.md [hunk 7]
```diff
@@ -43,6 +39,7 @@ modules for machine learning tasks.
     -   Count the occurrences of words in all the documents to calculate
         [TF-IDF](https://en.wikipedia.org/wiki/Tf%E2%80%93idf)
         weights.
+    -  Generate [embeddings](https://en.wikipedia.org/wiki/Embedding) on text data using LLMs.
```
Comment (rszper) [inline website/www/site/content/en/documentation/ml/preprocess-data.md:42]:
```suggestion
    -  Generate [embeddings](https://en.wikipedia.org/wiki/Embedding) on text data using large language models (LLMs).
```

File: website/www/site/content/en/documentation/ml/preprocess-data.md [hunk 8]
```diff
@@ -70,9 +70,13 @@ TensorFlow documentation.
 | TFIDF | See [`tft.tfidf`](https://www.tensorflow.org/tfx/transform/api_docs/python/tft/tfidf) in the TensorFlow documentation. |:
 {{< /table >}}
 
-Apply the transforms on either single or multiple columns passed as a
-`dict` on structured data. Keys are column names and values are lists containing
-each column's data.
+### Generate Text Embeddings
+
```
Comment (rszper) [inline website/www/site/content/en/documentation/ml/preprocess-data.md:74]: Add a sentence after the heading but before the table:

You can use `MLTranfrorm` to generate embeddings that you can use to push data into vector databases or to run inference.

Comment (AnandInguva) [reply website/www/site/content/en/documentation/ml/preprocess-data.md:78]: Done

File: website/www/site/content/en/documentation/ml/preprocess-data.md [hunk 9]
```diff
@@ -45,18 +39,23 @@ modules for machine learning tasks.
     -   Count the occurrences of words in all the documents to calculate
         [TF-IDF](https://en.wikipedia.org/wiki/Tf%E2%80%93idf)
         weights.
+    -  Generate [embeddings](https://en.wikipedia.org/wiki/Embedding) on text data using large language models (LLMs).
```
Comment (rszper) [inline website/www/site/content/en/documentation/ml/preprocess-data.md:42]: We might want to make this the first list item, depending on how much we want to emphasize embeddings. We could also move the Text embedding transforms section before the Data processing transforms that use TFT section.

File: website/www/site/content/en/documentation/ml/preprocess-data.md [hunk 10]
```diff
@@ -45,18 +39,23 @@ modules for machine learning tasks.
...
 
 ## Support and limitations {#support}
 
--   Available in the Apache Beam Python SDK versions 2.50.0 and later.
--   Supports Python 3.8 and 3.9.
+-   Available in the Apache Beam Python SDK versions 2.53.0 and later.
+-   Supports Python 3.8, 3.9 and 3.10.
```
Comment (rszper) [inline website/www/site/content/en/documentation/ml/preprocess-data.md:47]:
```suggestion
-   Supports Python 3.8, 3.9, and 3.10.
```

File: website/www/site/content/en/documentation/ml/preprocess-data.md [hunk 11]
```diff
@@ -45,18 +39,23 @@ modules for machine learning tasks.
...
 -   Only available for pipelines that use [default windows](/documentation/programming-guide/#single-global-window).
--   Only supports one-to-one transform mapping on a single element.
 
 ## Transforms {#transforms}
 
-You can use `MLTransform` to perform the following data processing transforms.
-For information about the transforms, see
+You can use `MLTransform` to perform various data processing transforms.
```
Comment (rszper) [inline website/www/site/content/en/documentation/ml/preprocess-data.md:52]:
```suggestion
You can use `MLTransform` to generate text embeddings and to perform various data processing transforms.
```

File: website/www/site/content/en/documentation/ml/preprocess-data.md [hunk 12]
```diff
@@ -197,32 +198,3 @@ Replace the following values:
...
-
-```
-data_pcoll | MLTransform(write_artifact_location=<LOCATION>).with_transform(ScaleTo01(columns=['x']))
-```
-
-The `ScaleTo01` transformation produces two artifacts: the `min` and the `max`
-of the entire dataset. For more information, see the
-[Artifacts](#artifacts) section on this page.
-
-## Metrics {#metrics}
```
Comment (rszper) [inline website/www/site/content/en/documentation/ml/preprocess-data.md:218]: Just want to verify whether removing the Metrics section is intentional.

File: website/www/site/content/en/documentation/ml/preprocess-data.md [hunk 13]
```diff
@@ -73,18 +72,20 @@ TensorFlow documentation.
 | TFIDF | See [`tft.tfidf`](https://www.tensorflow.org/tfx/transform/api_docs/python/tft/tfidf) in the TensorFlow documentation. |:
 {{< /table >}}
 
-Apply the transforms on either single or multiple columns passed as a
-`dict` on structured data. Keys are column names and values are lists containing
-each column's data.
+### Text embedding transforms
+
+You can use `MLTranfrorm` to generate embeddings that you can use to push data into vector databases or to run inference.
```
Comment (rszper) [inline website/www/site/content/en/documentation/ml/preprocess-data.md:77]:
```suggestion
You can use `MLTranform` to generate embeddings that you can use to push data into vector databases or to run inference.
```

**Review by rszper — APPROVED** [review]

**Pull request merged**